File: app/src/main/java/me/gabeg/sicksends/sport/SsSportViewModel.java
package me.gabeg.sicksends.sport;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemViewModel;

/**
 * Sport view model.
 */
public class SsSportViewModel
	extends SsProblemViewModel<SsSportRepository, SsSport>
{

	/**
	 */
	public SsSportViewModel(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemViewModel#buildRepository(Application)
	 */
	public void buildRepository(Application app)
	{
		this.mRepository = new SsSportRepository(app);
	}

}

File: app/src/main/java/me/gabeg/sicksends/sport/SsSportViewModel.kt
package me.gabeg.sicksends.sport

import androidx.lifecycle.SavedStateHandle
import dagger.hilt.android.lifecycle.HiltViewModel
import me.gabeg.sicksends.problem.SsGenericProblemViewModel
import javax.inject.Inject

/**
 * Sport view model.
 */
@HiltViewModel
class SsSportViewModel @Inject constructor(
	savedStateHandle: SavedStateHandle,
	repo : SsSportRepository)
		: SsGenericProblemViewModel<SsSportProblem>(savedStateHandle, repo)
